libs/react-hooks/src/use-analytics/use-analytics.ts

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useMemo, useState } from 'react'
+import { snakeCase } from 'lodash'
+import ReactGA from 'react-ga4'
+import { InitOptions } from 'react-ga4/types/ga4'
+import { UserData } from '@globalfishingwatch/api-types'
+
+export enum TrackCategory {
+  User = 'user',
+}
+
+export type TrackEventParams<T> = {
+  category: T | TrackCategory
+  action: string
+  label?: string
+  value?: any
+}
+
+export const trackEvent = <T>({ category, action, label, value }: TrackEventParams<T>) => {
```

Comment:
👏 love the way of reusing generics [here](https://github.com/GlobalFishingWatch/frontend/pull/2125/files#diff-21064761a7704a322805ab08d7ef25cfd10b406974c3a5a2862a4452941ed1f3R27)